python/graphstorm/gconstruct/remap_result.py

```diff
@@ -0,0 +1,477 @@
+from torch import multiprocessing
+from torch.multiprocessing import Process
+from ..model.utils import pad_file_index
+from .file_io import write_data_parquet
+from .id_map import IdReverseMap
+from ..utils import get_log_level
+
+from ..config import (GSConfig,
+                      get_argument_parser,
+                      BUILTIN_TASK_EDGE_CLASSIFICATION,
+                      BUILTIN_TASK_EDGE_REGRESSION,
+                      BUILTIN_TASK_NODE_CLASSIFICATION,
+                      BUILTIN_TASK_NODE_REGRESSION)
+
+id_maps = {}
+
+def worker_remap_edge_pred(pred_file_path, src_nid_path,
+    dst_nid_path, src_type, dst_type,
+    output_fname_prefix, chunk_size, preserve_input):
+    """ Do one edge remapping task
+
+        Parameters
+        ----------
+        pred_file_path: str
+            The path to the prediction result.
+        src_nid_path: str
+            The path to the file storing source node ids
+        dst_nid_path: str
+            The path to the file storing destination node ids
+        src_type: str
+            Src node type.
+        dst_type: str
+            Dst node type.
+        output_fname_prefix: str
+            Output file name prefix.
+        chunk_size: int
+            Max number of raws per output file.
+        preserve_input: bool
+            Whether the input data should be removed.
+    """
+    pred_result = th.load(pred_file_path).numpy()
+    src_nids = th.load(src_nid_path).numpy()
+    dst_nids = th.load(dst_nid_path).numpy()
+    src_id_map = id_maps[src_type]
+    dst_id_map = id_maps[dst_type]
+    num_chunks = math.ceil(len(pred_result) / chunk_size)
+    for i in range(num_chunks):
+        output_fname = f"{output_fname_prefix}_{pad_file_index(i)}.parquet"
+
+        start = i * chunk_size
+        end = (i + 1) * chunk_size if i + 1 < chunk_size else len(pred_result)
+        pred = pred_result[start:end]
+        src_nid = src_id_map.map_id(src_nids[start:end])
+        dst_nid = dst_id_map.map_id(dst_nids[start:end])
+        data = {"pred": pred,
+                "src_nid": src_nid,
+                "dst_nid": dst_nid}
+
+        write_data_parquet(data, output_fname)
+
+    if preserve_input is False:
+        os.remove(pred_file_path)
+        os.remove(src_nid_path)
+        os.remove(dst_nid_path)
+
+def worker_fn(worker_id, task_queue, func):
+    """ Process remap tasks with multiprocessing
+
+        Parameters
+        ----------
+        worker_id: int
+            Worker id.
+        task_queue: Queue
+            Task queue.
+        func: function
+            Function to be executed.
+    """
+    try:
+        while True:
+            # If the queue is empty, it will raise the Empty exception.
+            idx, task_args = task_queue.get_nowait()
+            logging.debug("worker %d Processing %s task", worker_id, idx)
+            func(**task_args)
+    except queue.Empty:
+        pass
+
+def multiprocessing_remap(tasks, num_proc, remap_func):
```

Comment:
should we make this a standardized utility function for multiprocessing?

Reply:
You mean we move it into utils.py?
In the following PRs, both remap_node_predictions and remap_node_embeddings will use this function.

Reply:
i think so. we might want to generalize it for other purpose as well. for example, gconstruct has a multi-processing function as well.

Reply:
OK. I moved it into utils.py as named it multiprocessing_exec_no_return.